Can't build Dockerfile - " /bin/sh: 1: gx: Exec Format Error"

#### Version information: Cloned from Repo

#### Type: Bug - Build

#### Description:

docker build .
/bin/sh: 1: gx: Exec format error
make: *** [gx-deps] Error 2


I have been trying to figure out what is triggering this in gx.mk to no avail - the files pulled down in bin/gx run from the CLI (they also do not appear to be a shell file).

I'm open to ideas...



# Go Environment
Nicks-MacBook-Pro:go-ipfs njl$ go version
go version go1.10.2 darwin/amd64

#Envirionment Docker
:go-ipfs njl$ docker version
Client:
 Version:      18.03.1-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   9ee9f40
 Built:        Thu Apr 26 07:13:02 2018
 OS/Arch:      darwin/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.03.1-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.5
  Git commit:   9ee9f40
  Built:        Thu Apr 26 07:22:38 2018
  OS/Arch:      linux/amd64
  Experimental: true

Root Cause: This was because OSX Docker needs gx and gx-go to be a linux binary. 

I ended up building both / copying them in place to get past this issue.

Leaving me with the next set of errors...
`o build -i -ldflags="-X "github.com/ipfs/go-ipfs/repo/config".CurrentCommit=2ba7179" -o "cmd/ipfs/ipfs" "github.com/ipfs/go-ipfs/cmd/ipfs"
Putting child 0x5643dc850f00 (cmd/ipfs/ipfs) PID 1034 on the chain.
Live child 0x5643dc850f00 (cmd/ipfs/ipfs) PID 1034 
fuse/mount/fuse.go:12:2: cannot find package "bazil.org/fuse" in any of:
	/go/src/github.com/ipfs/go-ipfs/vendor/bazil.org/fuse (vendor tree)
	/usr/local/go/src/bazil.org/fuse (from $GOROOT)
	/go/src/bazil.org/fuse (from $GOPATH)
fuse/mount/fuse.go:13:2: cannot find package "bazil.org/fuse/fs" in any of:
	/go/src/github.com/ipfs/go-ipfs/vendor/bazil.org/fuse/fs (vendor tree)
	/usr/local/go/src/bazil.org/fuse/fs (from $GOROOT)
	/go/src/bazil.org/fuse/fs (from $GOPATH)
cmd/ipfs/main.go:28:2: cannot find package "github.com/Kubuxu/go-os-helper" in any of:
	/go/src/github.com/ipfs/go-ipfs/vendor/github.com/Kubuxu/go-os-helper (vendor tree)
	/usr/local/go/src/github.com/Kubuxu/go-os-helper (from $GOROOT)
	/go/src/github.com/Kubuxu/go-os-helper (from $GOPATH)
unixfs/hamt/hamt.go:32:2: cannot find package "github.com/Stebalien/go-bitfield" in any of:
	/go/src/github.com/ipfs/go-ipfs/vendor/github.com/Stebalien/go-bitfield (vendor tree)
	/usr/local/go/src/github.com/Stebalien/go-bitfield (from $GOROOT)
	/go/src/github.com/Stebalien/go-bitfield (from $GOPATH)
namesys/proquint.go:8:2: cannot find package "github.com/bren2010/proquint" in any of:
	/go/src/github.com/ipfs/go-ipfs/vendor/github.com/bren2010/proquint (vendor tree)
	/usr/local/go/src/github.com/bren2010/proquint (from $GOROOT)
	/go/src/github.com/bren2010/proquint (from $GOPATH)
exchange/reprovide/reprovide.go:10:2: cannot find package "github.com/cenkalti/backoff" in any of:
	/go/src/github.com/ipfs/go-ipfs/vendor/github.com/cenkalti/backoff (vendor tree)
	/usr/local/go/src/github.com/cenkalti/backoff (from $GOROOT)
	/go/src/github.com/cenkalti/backoff (from $GOPATH)
`

So I had been getting the following error attempting to build the ipfs-cluster container on macOS:
```
deptools/gx_v0.13.0 install --global
deptools/gx_v0.13.0: 1: deptools/gx_v0.13.0: ����: not found
deptools/gx_v0.13.0: 2: deptools/gx_v0.13.0: Syntax error: word unexpected (expecting ")")
Makefile:63: recipe for target 'deps' failed
make: *** [deps] Error 2
```
And going off what you said @NickJLange, I replaced the macOS gx/gx-go binaries and the build is now working 👍

--- a/.dockerignore
+++ b/.dockerignore
@@ -5,3 +5,9 @@ Dockerfile.fast
 !.git/refs/
 !.git/packed-refs
 test/sharness/lib/sharness/
+
+# The Docker client might not be running on Linux
+# so delete any compiled binaries
+bin/gx
+bin/gx*
+bin/tmp

